lib/book_settings.py

```diff
@@ -190,7 +191,8 @@ def search_for_goodreads(self, keywords):
         if not urlsearch:
             return None
 
-        return 'https://www.goodreads.com' + urlsearch.group(1)
+        url = 'https://www.goodreads.com' + urlsearch.group(1)
+        return urlparse.urljoin(url, urlparse.urlparse(url).path)
```

Comment:
Better and clearer IMO: `urlparse.urlparse(url)._replace(query=None).geturl()`
[ignore the underscore in `_replace`, it is a public function, just named to avoid conflicting with field names, as per the documentation]

```
>>> print urlparse.urlparse(url).geturl()
https://www.goodreads.com/some/think?some=arg&other=another
>>> print urlparse.urlparse(url)._replace(query=None).geturl()
https://www.goodreads.com/some/think
>>> 
```